dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -149,30 +151,32 @@ export default function Catalog() {
         cluster,
         namespace,
         reposFilter,
-        nextPageToken,
+        localNextPageToken,
```

Comment:
Why renaming the variable? Why is important the nuance of "local"?

Reply:
Becasue `nextPageToken` is coming from the global packages state and `localNextPageToken` is a copy of this value, but in the component's state. This way, we have fine-grained control of the lifecycle of the state. 
In fact, this is the crux of the PR: `nextPageToken` gets updated each time new data arrives (that's why it was failing).
 So, technically we are not renaming but storing a local copy of `nextPageToken` and managing how it should be updated (that is, on each pagination event).